tests/psbt.rs

```diff
@@ -0,0 +1,459 @@
+    let expected_psbt_hex = include_str!("data/combine_psbt_hex");
+    let expected_psbt = hex_psbt!(expected_psbt_hex).unwrap();
+
+    this.combine(that).expect("failed to combine PSBTs");
+
+    assert_eq!(this, expected_psbt);
+    this
+}
+
+/// Does the finalize step according to the BIP, returns the combined PSBT. Verifies against BIP174
+/// test vector.
+fn finalize(psbt: Psbt) -> Psbt {
+    let expected_psbt_hex = include_str!("data/finalize_psbt_hex");
+    let expected_psbt = hex_psbt!(expected_psbt_hex).unwrap();
+
+    let psbt = finalize_psbt(psbt);
+
+    assert_eq!(psbt, expected_psbt);
+    psbt
+}
+
+/// Does the transaction extractor step according to the BIP, returns the combined PSBT. Verifies
+/// against BIP174 test vector.
+fn extract_transaction(psbt: Psbt) -> Transaction {
+    let expected_tx_hex = include_str!("data/extract_tx_hex");
+
+    let tx = psbt.extract_tx();
+    
+    let got = serialize_hex(&tx);
+    assert_eq!(got, expected_tx_hex);
+
+    tx
+}
+
+/// Combines two PSBTs lexicographically according to the BIP. Verifies against BIP174 test vector.
+fn combine_lexicographically() {
+    let psbt_1_hex = include_str!("data/lex_psbt_1_hex");
+    let psbt_2_hex = include_str!("data/lex_psbt_2_hex");
+
+    let expected_psbt_hex = include_str!("data/lex_combine_psbt_hex");
+    let expected_psbt = hex_psbt!(expected_psbt_hex).unwrap();
+    
+    let v = Vec::from_hex(&psbt_1_hex).unwrap();
+    let mut psbt_1: Psbt = deserialize(&v).expect("failed to deserialize psbt 1");
+
+    let v = Vec::from_hex(&psbt_2_hex).unwrap();
+    let psbt_2: Psbt = deserialize(&v).expect("failed to deserialize psbt 2");
+
+    psbt_1.combine(psbt_2).expect("failed to combine PSBTs");
+    
+    assert_eq!(psbt_1, expected_psbt);
+}
+
+/// Signs `psbt` with `keys` if required.
+fn sign(mut psbt: Psbt, keys: &[PrivateKey]) -> Psbt {
+    let secp = SECP256K1;
+
+    let mut inputs = psbt.inputs.clone();
+
+    for (i, input) in inputs.iter_mut().enumerate() {
+        let required_pks: Vec<_> = input.bip32_derivation.keys().collect();
+        for sk in keys {
+            let pk = sk.public_key(&secp);
+            if required_pks.contains(&&pk.inner) {
+                psbt.sign(&sk, i, secp).expect("failed to sign input with key");
+            }
+        }
+    }
+
+    psbt
+}
+
+/// Finalizes a PSBT accord to the Input Finalizer role described in BIP174.
+fn finalize_psbt(mut psbt: Psbt) -> Psbt {
+    use bitcoin::util::psbt::serialize::Serialize;
+
+    // Input 0: legacy UTXO
+
+    // TODO: Remove this explicit scope once we bump MSRV. The Rust 1.29 borrow checker is not
+    // sophisticated enough to handle the immutable and mutable borrow of psbt.inputs[0].
+    let script_sig = {
+        let sigs: Vec<_> = psbt.inputs[0].partial_sigs.values().collect();
+        script::Builder::new()
+            // TODO: Consider adding OP_0 alias to rust-bitcoin (`pub static OP_0: All = all::OP_PUSHBYTES_0;`).
```

Comment:
We got the greenlight on this already: https://github.com/rust-bitcoin/rust-bitcoin/pull/935#discussion_r854539416

Before merging we should either put an initial patch before this patch that adds the new alias or raise an issue describing what needs to be done (my preference would be the first option). Either way we should remove all the TODOs from this file before attempting to merge (I just wrote them in there for us during development).